### PR TITLE
DEP: Deprecate (rather than remove) the int-via-float parsing in loadtxt

### LINKED ISSUE #21652: BUG?:  `loadtxt` is now stricter with integers
Scikit-image noticed this (not sure if it is in a particularly important part): https://github.com/scikit-image/scikit-image/issues/6397  CC @hmaarrfk

There is a point that `np.savetxt` doesn't pair well (because it defaults to storing floats with `%e18`, which is a bit weird).  But we also had a good reason to try it:  The old parser used `int(float(value))` which is pretty terrible, as it for example loses bits for large numbers.

We could probably try to make it a deprecation instead, at least if it comes up more.  Working around it is pretty easy for end-users IMO (in most cases `converters=float` works on new NumPy versions non older you would have to use casting probably `np.loadtxt(...).astype(int)`).

### PR DESCRIPTION
**Pull request by seberg**
Note that previously `float()` was used, so that we still lose the
capability of parsing underscores.  I think we should just live with
that.
This is mainly to allow a smoother transition, since:

   from skimage import feature

happens to run into it (and other libraries might as well in principle)

---

IMO, this should be good enough to close the issue.  I do think we should go with this as a deprecation right now.  **This is draft, since I would like a link to a brief docs page.** @rossbar do you think you may have time to look into that?  Otherwise, I will think about it a bit.

~(Whoops, plus tests are still a TODO at this time :), but will do that right away.).~

Closes gh-21652

### CODE REVIEW

File: numpy/core/src/multiarray/textreading/str_to_int.c [hunk 1]
```diff
@@ -41,8 +63,24 @@
...
+            }                                                               \
+            if (!pconfig->gave_int_via_float_warning) {                     \
+                pconfig->gave_int_via_float_warning = true;                 \
+                if (PyErr_WarnEx(PyExc_DeprecationWarning,                  \
+                        deprecation_msg, 3) < 0) {                          \
+                    return -1;                                              \
+                }                                                           \
+            }                                                               \
+            pconfig->gave_int_via_float_warning = true;                     \
+            x = (uintw##_t)fval;                                            \
```
Comment (seberg) [inline numpy/core/src/multiarray/textreading/str_to_int.c:83]: Now that we have C++, this should even more be cleaned up and stop being a macro.  But I don't think it is important enough to try that now.  Especially on something that needs backporting.

**Comment by rossbar** [comment]
>  @rossbar do you think you may have time to look into that?

Yes, though I'll be out for the next few days! Don't wait on me for anything time-sensitive, I can definitely look when I'm back at the computer

**Comment by charris** [comment]
The failing test can be ignored.

> Should we merge this, and I create a backport PR directly editing the release note?

Sounds good. I'll put this in now. Thanks Sebastian.

**Comment by hmaarrfk** [comment]
Thanks all!

**Pull request merged**